src/Filesystem/F.php

```diff
@@ -846,6 +846,18 @@ public static function typeToExtensions(string $type): ?array
 		return static::$types[$type] ?? null;
 	}
 
+	/**
+	 * Safest way of PHP's native unlink function
+	 */
+	public static function unlink(string $file): bool
+	{
+		try {
+			return unlink($file);
+		} catch (Exception $e) {
```

Comment:
Have you verified that catching `Exception` works? AFAIK, a PHP warning is not an exception. Maybe we need to use `set_error_handler()` and `restore_error_handler()` instead (unless they changed the PHP behavior).

Reply:
I've tested on my local machine and unit tests. `try catch` seems works great! 🥳 🤷‍♂️ 

- file exists case ✅ 
- no file case ✅ 
- file exists with no permission ✅

Reply:
When I removed the `try catch` block, throws exception (debug enabled). No error on debug mode disabled.

Reply:
I have an idea about that I think `whoops` handle this. Is it possible? Because `Whoops\Exception\ErrorException` throws when `unlink` fails.

Reply:
The PR seems ready when we are sure about try catch in unlink error handler.

Reply:
Yes, I agree that this must be because of Whoops. If I set the `whoops` option to `false`, it doesn't work. So I think we should really use an error handler instead for maximum compatibility.